[FEATURE] Add test to gate v1 epic 1

billdirks opened the pull request:
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

billdirks committed:
Add test to gate v1 epic 1

billdirks commented on tests/v1_demos/test_batch_validate.py:
```diff
@@ -0,0 +1,30 @@
+    batch = context.sources.pandas_default.read_csv(
+        "https://raw.githubusercontent.com/great-expectations/gx_tutorials/main/data/yellow_tripdata_sample_2019-01.csv"
+    )
+    expectation = gxe.ExpectColumnValuesToNotBeNull(
```
Currently dies here with error:
`AttributeError: module 'great_expectations.expectations' has no attribute 'ExpectColumnValuesToNotBeNull'`

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=9127&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/9127/a5b06622/5592dc96379433b3e3066d68caebfad23da2bd01.svg)](https://app.codesee.io/r/reviews?pr=9127&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

billdirks committed:
Add postgresql test.

billdirks commented on tests/v1_demos/test_batch_validate.py:
```diff
@@ -0,0 +1,56 @@
+
+
+def _get_sql_batch(context: AbstractDataContext) -> Batch:
+    return context.sources.pandas_default.read_sql(
```
I will need to investigate how to populate this database.

billdirks committed:
Update test_batch_validate.py

Remove csv from batch validate test function now that the sources are parameterized.

billdirks committed:
Merge branch 'develop' into gate_v1_epic_1

billdirks committed:
Merge branch 'develop' into gate_v1_epic_1

billdirks commented on tests/v1_demos/test_batch_validate.py:
```diff
@@ -0,0 +1,56 @@
+    return request.param
+
+
+def test_batch_validate(get_batch: Callable[[AbstractDataContext], Batch]):
```
Another possible cleaner strategy here is to put this code in a helper function. Then have 2 tests, one for csv and one for sql that calls into this helper function. This would be more explicit with markers and involve less indirection.

tyler-hoffman replied:
Is there a way to get this working with just `parametrize` and the fixtures/marks? That feels like it'd be the simplest to read.

billdirks committed:
Merge branch 'develop' into gate_v1_epic_1

billdirks committed:
Merge branch 'develop' into gate_v1_epic_1

billdirks commented on tests/v1_demos/test_batch_validate.py:
```diff
@@ -0,0 +1,56 @@
+    batch = get_batch(context)
+    expectation = gxe.ExpectColumnValuesToNotBeNull(
+        "pu_datetime",
+        notes="These are filtered out upstream, because the entire record is garbage if there is no pu_datetime",
```
Currently dies with
```
get_batch = <function _get_csv_batch at 0x7fc5abc595e0>

    def test_batch_validate(get_batch: Callable[[AbstractDataContext], Batch]):
        context = gx.get_context()
        batch = get_batch(context)
>       expectation = gxe.ExpectColumnValuesToNotBeNull(
            "pu_datetime",
            notes="These are filtered out upstream, because the entire record is garbage if there is no pu_datetime",
        )
E       TypeError: __init__() takes 1 positional argument but 2 were given

tests/v1_demos/test_batch_validate.py:41: TypeError
```

billdirks committed:
explicitly use kwarg 'column'

billdirks committed:
Temporarily comment out sql test.

billdirks committed:
Update asserts to match data file.

tyler-hoffman reviewed (approved):
This is exciting. I think it's a good plan to merge this as is and circle back on the SQL piece.

billdirks commented:
Now that we have https://github.com/great-expectations/great_expectations/pull/9176/files about to merge, I'm going to close this one.
